.github/workflows/builder.yaml

```diff
@@ -0,0 +1,57 @@
+name: CI Build & Push Builder Image
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+    - Dockerfile.builder
+    - .bazelversion
+
+jobs:
+  build-and-push-builder:
+    timeout-minutes: 360
+    name: Build and push multi-arch builder images
+    runs-on: ubuntu-latest-64-cores-256gb
```

Comment:
This build can't significantly benefit from the large runner, so maybe scale back to "ubuntu-latest"?